synapse/handlers/sso.py

```diff
@@ -810,6 +817,10 @@ def is_allowed_mime_type(content_type: str) -> bool:
                 str(avatar_mxc_url),
             )
 
+            # delete previous avatar, if one existed
+            if delete_previous_avatar and existing_media_id is not None:
```

Comment:
(`existing_media_id` seems to be the 'local' part of the media ID as far as I can tell, but we haven't checked whether it is media for our server name.)